prov/shm/src/smr_progress.c

```diff
@@ -1014,10 +1024,12 @@ void smr_ep_progress(struct util_ep *util_ep)
 
 	ep = container_of(util_ep, struct smr_ep, util_ep);
 
-	smr_progress_resp(ep);
-	smr_progress_cmd(ep);
-
-	smr_progress_sar_list(ep);
+	if (!ep->region->signal) {
+		smr_signal_reset(ep->region);
+		smr_progress_resp(ep);
+		smr_progress_cmd(ep);
+		smr_progress_sar_list(ep);
+	}
```

Comment:
It seems that this code should acquire the signal lock before reading it.  The value of signal also seems opposite from what I expected.  Signal is acting as a bool, but false indicates the signaled state.

Based on the changes, an atomic value here would be better.

Reply:
Fair. Better?

Reply:
Better.  Best would be:
```
if (ofi_atomic_cas_bool32(&ep->region->signal, 1, 0)) {
    smr_progress_resp(ep);
    smr_progress_cmd(ep);
    smr_progress_sar_list(ep);
}
```

Reply:
👍 Updated